### ISSUE TITLE
Feature Filter by path

### ISSUE THREAD

**Issue by workgroupengineering** [AvaloniaUI/Avalonia.NameGenerator#26]
ince https://github.com/AvaloniaUI/Avalonia.NameGenerator/releases/tag/0.2.0-preview the  the generator processes all  classes inheriting from the Avalonia.INamed interface. 

## Current behavior
 the generator processes all  classes inheriting from the Avalonia.INamed interface. 
## New behavior
When you define `AvaloniaNameGeneratorFilterByPath`, the generator only processes classes that inherit from the Avalonia.INamed interface with mattch specified glob pattern.

####  It processes all classes with glob pattern
```xml
<PropertyGroup>
   <AvaloniaNameGeneratorFilterByPath>Presentation/Views/*View.xaml</AvaloniaNameGeneratorFilterByPath>
</PropertyGroup>
```

#### It processes all classes with multiple glob pattern
```xml
<PropertyGroup>
   <AvaloniaNameGeneratorFilterByPath>*View.axaml;*Control.axaml</AvaloniaNameGeneratorFilterByPath>
</PropertyGroup>
```

**Comment by worldbeater** [comment]
Just out of curiosity, how the `x:Name` processing is handled in WPF/WinForms worlds? I don't remember namespace/path exclusions there, their generator just works. Probably we should process only those classes that have the appropriate XAML file with the same name placed nearby without emitting the warnings (and silently skip that file if no XAML file found)? I'd prefer not to overcomplicate such a straightforward thing as `x:Name` reference generation. Also assuming the additional XAML file existence check is relatively cheap.

**Comment by robloo** [comment]
I think two criteria would work to only generate name field references when needed:

 1. Only processes XAML (.xaml or .axaml) files when there is a corresponding file of the same name with *.xaml.cs or *.axaml.cs
 2. Only generate name references when the XAML actually contains "Name" or "x:Name" attributes

**Comment by worldbeater** [comment]
Hmm also as @rstm-sf noted there is yet another extension we should support — `paml` https://github.com/AvaloniaUI/Avalonia/search?q=paml

**Comment by worldbeater** [comment]
Probably a more robust solution would be to:
1. Parse XAML files included as `<AdditionalFile />`s
2. Skip files that has no `x:Class` directive at the root node
3. Resolve the type from the `x:Class` directive via our MiniCompiler
4. Perform the codegen

**Comment by robloo** [comment]
paml extension files are very old from before the project was named Avalonia. It probably should NOT be supported to encourage others to update.

Using x:Class is a great idea I forgot about.

**Comment by workgroupengineering** [comment]
> 
> 
> Probably a more robust solution would be to:
> 
>     1. Parse XAML files included as `AdditionalFile`
> 
>     2. Skip files that have no `x:Class` directive
> 
>     3. Resolve the type from the `x:Class` directive via our MiniCompiler, RoslynTypeSystem and XamlX
> 
>     4. Perform the codegen
> 
> 
> This implementation won't suffer from errors like "uh-oh there is no XAML, PAML, or AXAML file nearby" and also won't process irrelevant files. This way we can remove the AXN0001 warning and make the AXN0003 warning an error.

it's a great idea, but xml parsing is expensive, having optional filters that have to limit the scope would improve its effectiveness.

**Comment by worldbeater** [comment]
Yeah since we are now parsing `x:Class` limiting the scope makes sense.

**Comment by RomanSoloweow** [comment]
It would be cool to set different generator behavior for different file types

```XML
<AvaloniaNameGeneratorFilterByPath Behavior="InitialComponent">View.axaml;Control.axaml</AvaloniaNameGeneratorFilterByPath>
<AvaloniaNameGeneratorFilterByPath Behavior="OnlyProperties">MyView.axaml;MyControl.axaml</AvaloniaNameGeneratorFilterByPath>
```

**Comment by worldbeater** [comment]
Closed by https://github.com/AvaloniaUI/Avalonia.NameGenerator/pull/32, shipped to NuGet as 0.4.1, see also [Advanced Usage](https://github.com/AvaloniaUI/Avalonia.NameGenerator#advanced-usage).
